### PR TITLE
Fix invalid assert.throws statements

### LINKED ISSUE #416: Fix some invalid assert.throws statements
While writing my tests for #412, I detected some wrong assert.throws statements. I can fix these after I finish the #412 issue (which is almost ready)

These invalid statements look like this:
```js
assert.throws(() => {
  new dfd.NDframe({ data, isSeries: false, columns: [ "A", "A", "C" ] }),
  Error,
  "ColumnIndexError: Column index must contain unique values";
});
```

and they should be like this:
```js
assert.throws(
  () => new dfd.NDframe({ data, isSeries: false, columns: [ "A", "A", "C" ] }),
  Error,
  "ColumnIndexError: Column index must contain unique values"
);
```

I checked it and these wrong tests are passing even if the error message does not match. Fortunately, these tests will fail if no error is thrown during in the execution, so at least is guaranteed that an error will be thrown (but it does not check which type of error).

Thanks!

### PR DESCRIPTION
**Pull request by igonro**
Closes #416.

- [x] Fix invalid `assert.throws` statements
- [x] Run the tests again and check that the tests passed

To review: @risenW @steveoni

### CODE REVIEW

**Comment by igonro** [comment]
After fixing the invalid `assert.throws`, some of the node tests are not passing. This is the output:

```
  707 passing (965ms)
  2 failing

  1) DataFrame
       addColumn
         throw error for wrong column length:

      AssertionError: expected [Function] to throw error including 'ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 4' but got 'ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 3'
      + expected - actual

      -ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 3
      +ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 4

      at Context.<anonymous> (test/core/frame.test.ts:131:20)
      at processImmediate (internal/timers.js:464:21)

  2) Generic (NDFrame)
       NDframe Created from Array
         Throws error on duplicate index:

      AssertionError: expected [Function] to throw error including 'IndexError: Row index must contain unique values' but got 'IndexError: You provided an index of length 3 but Ndframe rows has length of 2'
      + expected - actual

      -IndexError: You provided an index of length 3 but Ndframe rows has length of 2
      +IndexError: Row index must contain unique values

      at Context.<anonymous> (test/core/generic.test.ts:72:20)
      at processImmediate (internal/timers.js:464:21)
```

**Comment by igonro** [comment]
The same happens for some browser tests:
```
Chrome Headless 98.0.4758.102 (Linux x86_64) DataFrame addColumn throw error for wrong column length FAILED
        AssertionError: expected [Function] to throw error including 'ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 4' but got 'ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 3'
            at Context.<anonymous> (/tmp/7ad0f53a5cff4b52986a8d5fc061a919.browserify.js:151280:14)

Chrome Headless 98.0.4758.102 (Linux x86_64) Generic (NDFrame) NDframe Created from Array Throws error on duplicate index FAILED
        AssertionError: expected [Function] to throw error including 'IndexError: Row index must contain unique values' but got 'IndexError: You provided an index of length 3 but Ndframe rows has length of 2'
            at Context.<anonymous> (/tmp/7ad0f53a5cff4b52986a8d5fc061a919.browserify.js:154123:14)
```

**Comment by igonro** [comment]
I solved those errors. Now all the tests are passing! 🎉 

I'm glad I casually found those `assert.throws` typos while working in #417, because they were hiding a little bug in the `shared/errors.ts` file, and other little bug in the tests of "Throws error on duplicate index" 😄

File: src/danfojs-browser/tests/core/frame.test.js [hunk 1]
```diff
@@ -123,11 +123,9 @@ describe("DataFrame", function () {
       const data = { alpha: [ "A", "B", "C", "D" ], count: [ 1, 2, 3, 4 ], sum: [ 20.3, 30.456, 40.90, 90.1 ] };
       const df = new dfd.DataFrame(data);
 
-      assert.throws(function () {
-        df.addColumn("new_column", new dfd.Series([ "a", "b", "c" ])),
-        Error,
-        'ParamError: Column data length mismatch. You provided data with length 3 but Ndframe has column of length 4';
-      });
+      assert.throws(
```
Comment (steveoni) [inline src/danfojs-browser/tests/core/frame.test.js:126]: @igonro can you apply lint fix here, to break into several lines

File: src/danfojs-browser/tests/core/generic.test.js [hunk 2]
```diff
@@ -59,20 +59,16 @@ describe("Generic (NDFrame)", function () {
...
 
     it("Throws error on duplicate index", function () {
-      let data = [ [ 21, 20, 1 ], [ 20, 25, 3 ] ];
-      assert.throws(() => {
-        new dfd.NDframe({ data, isSeries: false, index: [ 1, 1, 2 ] }),
-        Error,
-        "IndexError: Row index must contain unique values";
-      });
+      let data = [ [ 21, 20, 1 ], [ 20, 25, 3 ], [ 19, 30, 5 ] ];
+      assert.throws(
```
Comment (steveoni) [inline src/danfojs-browser/tests/core/generic.test.js:69]: Same here

**Review by steveoni — CHANGES_REQUESTED** [review]

**Comment by igonro** [comment]
Done! @steveoni 
I did it for the js files. Maybe is needed for the ts files too(?)
I looked at some assert.throws, but the lint style is a bit different sometimes.

**Pull request merged**